app/models/notification.rb

```diff
@@ -0,0 +1,34 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'RightSidePopup') }
+  scope :email, -> { where(type: 'EmailNotification') }
+
+  def self.types
+    %w(CenterPopup RightSidePopup EmailNotification)
+  end
+
+  def self.create(user, course, options = {})
+    case options['type']
+    when 'center_popup'
+      CenterPopup.create_notification(user, course, options)
```

Comment:
<code>create_notification</code> This function name looks a bit awkward, think about a better name ?